dagger-common/src/main/java/io/odpf/dagger/common/core/Transformer.java

```diff
@@ -5,4 +5,11 @@
  */
 public interface Transformer {
     StreamInfo transform(StreamInfo streamInfo);
+
+    /**
+     * Each created {@link DaggerContext} instance
+     */
+    static DaggerContext daggerContextGenerator() {
+        return DaggerContext.getInstance();
+    }
```

Comment:
I don't think this is the right place for this function. The methods of an interface should be indicative of its responsibility. Generating a dagger context object does not fall under the duties of a transformer,  even though it's concrete implementations may require it.

What do you think ?

Reply:
> The name of the method may be misleading. In this method we are getting the already initialised singleton instance of the DaggerContext

The concern is, a Transformer interface should not hold the signature of how to get a dagger context, because it does not relate to its sole role of transformation. Static interface methods are used to hold related utility methods together. However, getting a dagger context is not a requirement for transformation + not all transformer implementations will require it.

Additionally, think of this case. If some other class in the future also needs access to this DaggerContext object, we would need to keep adding these accessor methods either in the class or in the implemented interfaces. This will be cumbersome.

> Alternatively the instance of DaggerContext can be made available through constructor of implementing class. But for that need to touch lot of classes as mentioned [here](https://github.com/odpf/dagger/pull/202/files#r992288027).

Can you brainstorm on other approaches then ? For example, can we directly call `DaggerContext.getInstance()` from SQLTransformer to get the instance ?

Reply:
Making the `DaggerContext` object available to `SQLTransformer` class through constructor by replacing the `Configuration` parameter with `DaggerContext`. With this we can avoid two references of `configuration`  pointing to the same object. The flow of the context instance is KafkaProtoSQLProcessor -> StreamManager -> PostProcessorFactory -> ParentPostProcessor -> TransformProcessor.